ui/packages/atlasmap-core/src/models/config.model.ts

```diff
@@ -204,9 +199,12 @@ export class ConfigModel {
             '&collectionClassName=' + docInitModel.collectionClassName;
         }
       }
+    } else {
+      docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
     }
 
-    if (docInitModel.isSource) {
+    // somehow docInitModel.isSource gets always truthy... so explicitly compare with true
+    if (docInitModel.isSource === true) {
```

Comment:
That's strange - so if `docInitModel.isSource` is false it's still true?

Reply:
yep, not sure what caused it, as if the type (boolean) is lost at this point and recognized as string

Reply:
@gashcrumb @riccardo-forina do you know anything about this one? `docInitModel.isSource` is a typed boolean property and it is working fine on master, but after this refactoring `if(docInitModel.isSource)` evaluates to truthy even if it's false.

Reply:
nevermind, turned out it's caused by the type cast from JSON.parse() outcome into the contract interface. If the value in the JSON is string (e.g. {"isSource": "true"}) then a string value is delivered through a boolean property. It had to be cast to `string` anyway. crappy type system.